hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/strategies/VersionedEngagementsStrategy.java

```diff
@@ -72,19 +72,14 @@ public void init(Config defectDojoConfig, PersistenceProviderConfig persistenceP
 
   @Override
   public List<Finding> run(Scan scan, ScanFile scanResultFile) throws Exception {
-    Long userId = null;
 
-    if (this.config.getUserId() != null) {
-      LOG.debug("Using configured User Id");
-      userId = this.config.getUserId();
-    } else {
-      LOG.debug("Getting DefectDojo User Id via user profile API");
-      List<UserProfile> userProfiles = userProfileService.search();
-      if (userProfiles.isEmpty()) {
+    LOG.debug("Getting DefectDojo User Id via user profile API");
+    Long userId = null;
+    List<UserProfile> userProfiles = userProfileService.search();
+    if (userProfiles.isEmpty()) {
```

Comment:
Here you can use `.orElsethrow()` for a cleaner syntax: https://www.geeksforgeeks.org/optional-orelsethrow-method-in-java-with-examples/

Reply:
The return type of `search()` is of type `List<T>` not an `Optional`, so that is not possible.

Reply:
🤦‍♂️